qiita_pet/handlers/study_handlers/prep_template.py

```diff
@@ -77,3 +79,42 @@ def post(self):
         else:
             raise HTTPError(400, 'Unknown prep template action: %s' % action)
         self.write(result)
+
+
+class PrepFilesHandler(BaseHandler):
+    @authenticated
+    def get(self):
+        study_id = self.get_argument('study_id')
+        prep_file = self.get_argument('prep_file')
+        # atype = self.get_argument('type')
+
+        # TODO: Get file types for the artifact type
+        # FILE TYPE IN POSTION 0 MUST BE DEFAULT FOR SELECTED
+        file_types = ['raw_fwd', 'raw_rev']
+
+        selected = []
+        not_selected = []
+        _, base = get_mountpoint("uploads")[0]
+        uploaded = get_files_from_uploads_folders(study_id)
+        prep = pd.read_table(join(base, study_id, prep_file), sep='\t')
+        if 'run_prefix' in prep.columns:
+            # Use run_prefix column of prep template to auto-select
+            # per-sample uploaded files if available.
+            per_sample = True
+            prep_prefixes = set(prep['run_prefix'])
+            for _, filename in uploaded:
+                if splitext(filename)[0] in prep_prefixes:
```

Comment:
The issue there is then all the FWD and REV would be put into the forward seqs or whatever the default bin is. This default bin problem will need to be discussed during the meeting today.